qrenderdoc/Windows/PerformanceCounterViewer.cpp

```diff
@@ -0,0 +1,162 @@
+
+    case CounterUnit::Seconds: returnValue += QLatin1String{" s"}; break;
+
+    case CounterUnit::Absolute:
+    case CounterUnit::Ratio: break;
+  }
+
+  return returnValue;
+}
+
+PerformanceCounterViewer::PerformanceCounterViewer(ICaptureContext &ctx, QWidget *parent)
+    : QFrame(parent), ui(new Ui::PerformanceCounterViewer), m_Ctx(ctx)
+{
+  ui->setupUi(this);
+
+  m_Ctx.AddLogViewer(this);
+
+  connect(ui->captureCounters, &QToolButton::pressed, [=]() -> void {
+    PerformanceCounterSelection pcs(m_Ctx, this);
+    if(RDDialog::show(&pcs) == QDialog::Accepted)
+    {
+      const QList<GPUCounter> selectedCounters = pcs.GetSelectedCounters();
+
+      bool done = false;
+      m_Ctx.Replay().AsyncInvoke([=, &done](IReplayController *controller) -> void {
+        rdctype::array<GPUCounter> counters;
+        counters.create(selectedCounters.size());
+
+        QMap<GPUCounter, CounterDescription> counterDescriptions;
+
+        for(int i = 0; i < selectedCounters.size(); ++i)
+        {
+          counters[i] = (GPUCounter)selectedCounters[i];
+          counterDescriptions.insert(counters[i], controller->DescribeCounter(counters[i]));
+        }
+
+        QMap<GPUCounter, int> counterIndex;
+        for(int i = 0; i < selectedCounters.size(); ++i)
+        {
+          counterIndex.insert((GPUCounter)selectedCounters[i], i);
+        }
+
+        const auto results = controller->FetchCounters(counters);
+
+        GUIInvoke::call([this, results, counterDescriptions, counterIndex]() -> void {
+          ui->counterResults->clear();
+
+          QStringList headers;
+          headers << QLatin1String{"EID"};
+          for(const auto &cd : counterDescriptions)
+          {
+            headers << cd.name;
+          }
+
+          QMap<uint32_t, int> eventIdToRow;
+          for(const auto &result : results)
+          {
+            if(eventIdToRow.contains(result.eventID))
+              continue;
+            eventIdToRow[result.eventID] = eventIdToRow.size();
+          }
+
+          ui->counterResults->setColumnCount(headers.size());
+          ui->counterResults->setHorizontalHeaderLabels(headers);
+          ui->counterResults->setRowCount(eventIdToRow.size());
+          ui->counterResults->verticalHeader()->hide();
+
+          for(int i = 0; i < (int)results.size(); ++i)
+          {
+            int row = eventIdToRow[results[i].eventID];
+            ui->counterResults->setItem(row, 0,
+                                        new QTableWidgetItem(QString::number(results[i].eventID)));
+            ui->counterResults->setItem(row, counterIndex[results[i].counterID] + 1,
+                                        new QTableWidgetItem(FormatCounterResult(
+                                            results[i], counterDescriptions[results[i].counterID])));
+          }
+        });
+
+        done = true;
+      });
+
+      ShowProgressDialog(this, QLatin1String("Capturing counters"),
```

Comment:
Please use `tr()` for text like this that should be translated, and `lit()` for other literal strings like `'%'` or `"EID"` that don't need to be translated.